keps/draft-20180426-security-conformance.md

```diff
@@ -0,0 +1,682 @@
+  share internally with trusted users (e.g. an internal engineering team).
+
+  The criteria in this category must be satisfied for a Kubernetes cluster
+  to be considered secure at minimal.
+
+- `Restricted` (aka `Untrusted`):
+
+  In addition to `Essential`, it covers the threat sources of:
+  - Malicious workloads (not through Kubernetes API)
+
+  This profile limits the compatibility/capabilities to workloads to prevent
+  them from compromising the cluster.
+  It's still possible a malicious workload will compromise other workloads.
+  It's best for the use case to operate a secure Kubernetes cluster and share
+  with untrusted users without taking multi-tenancy into consideration.
+
+- `Isolated` (aka `Multi-tenancy`):
+
+  In addition to `Restricted`, it protects workloads from being compromised by
+  other workloads across the boundary of another tenant.
+
+  It's for the use case supporting multi-tenancy by sharing a Kubernetes cluster.
+
+- `Sandboxed`:
+
+  Similar to `Isolated`, but using a different container runtime for better isolation,
+  and possibly offering better compatibility and capabilities.
+  For example, using hypervisors, gVisor etc. which provides an additional layer of protection.
+
+Note: a workload is possible to be compromised if it's not properly
+designed/implemented according to security standards, or it's vulnerable.
+A compromised user identity can be used to compromise the whole system.
+These are out-of-scope of this proposal.
+
+The criteria described below is based on a namespace with default settings.
+Like a newly created namespace, with only service account `default`.
+Namespaces with special/privileged settings, like `kube-system`,
+or special namespaces required by some system components
+(e.g. nginx ingress controller) are not covered.
+
+Additional profiles can be proposed with combination of existing and new criteria.
+
+### Criteria
+
+#### Config File Permissions
+
+  *Conformance Profile: Essential*
+
+  The configuration files for API server, controller manager, scheduler and kubelet should have the correct setup of ownership and file permissions.
+  Without proper permission setup, a local user will be able to access prohibited information, including
+  - private keys
+  - secrets
+  - credentials
+
+  The criteria are:
+  - etcd data directory is owned and accessible only by the user running etcd;
+  - kubeconfig files for control plane is owned and accessible only by the user runs the control plane;
+  - private key files is owned and accessible only by the user who runs the process which consumes the key;
+  - kubeconfig for administration purpose (if present, e.g. `/etc/kubernetes/admin.conf`)
+    is owned and accessible only by the admin user (perm 0400)/group (perm 0440)
+    (the test doesn't verify particular user/group, but only the permissions).
+
+
+  Notes:
+  - the above statements exclude _root_ user who always have full access to the system.
```

Comment:
This statement isn't accurate. A root user running unconfined with full privileges in the "root" namespace has full access.

Reply:
Thanks!